rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -738,28 +738,22 @@ auto FleetUpdateHandle::Implementation::is_valid_assignments(
 }
 
 //==============================================================================
-std::size_t FleetUpdateHandle::Implementation::get_nearest_charger(
-  const rmf_traffic::agv::Planner::Start& start,
-  const std::unordered_set<std::size_t>& charging_waypoints)
+std::optional<std::size_t> FleetUpdateHandle::Implementation::get_nearest_charger(
+  const rmf_traffic::agv::Planner::Start& start)
 {
-  assert(!charging_waypoints.empty());
-  const auto& graph = (*planner)->get_configuration().graph();
-  Eigen::Vector2d p = graph.get_waypoint(start.waypoint()).get_location();
-
-  if (start.location().has_value())
-    p = *start.location();
+  if (charging_waypoints.empty())
+    return std::nullopt;
 
-  double min_dist = std::numeric_limits<double>::max();
-  std::size_t nearest_charger = 0;
+  double min_cost = std::numeric_limits<double>::max();
+  std::optional<std::size_t> nearest_charger = std::nullopt;
   for (const auto& wp : charging_waypoints)
   {
-    const auto loc = graph.get_waypoint(wp).get_location();
-    // TODO: Replace this with a planner call
-    // when the performance improvements are finished
-    const double dist = (loc - p).norm();
-    if (dist < min_dist)
+    const rmf_traffic::agv::Planner::Goal goal{wp};
+    const auto& planner_result = (*planner)->plan(start, goal);
```

Comment:
Since you're only using `ideal_cost` and not the actual plan that gets generated, you can use [`setup`](https://github.com/open-rmf/rmf_traffic/blob/f93fc723c39d8112f601877b3422d63296a0aed2/rmf_traffic/include/rmf_traffic/agv/Planner.hpp#L497-L500) here instead of `plan`. That will also give back the ideal cost but it won't require any planner iterations, so it should always be faster.

Reply:
Thanks for the suggestion!